teps/0095-repository-crd.md

```diff
@@ -0,0 +1,247 @@
+---
+status: proposed
+title: Repository CRD
+creation-date: '2021-11-19'
+last-updated: '2021-11-19'
+authors:
+- '@sbwsg'
+---
+
+# TEP-0095: Repository CRD
+---
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+Several projects and designs currently in-flight in the Tekton ecosystem
+are related to or directly working with source repository-related information.
+These include:
+
+- [Scheduled and Polling Runs in Tekton](0083-scheduled-and-polling-runs-in-tekton.md)
+- [Workflows](../working-groups.md#workflows)
+- [Pipeline-as-Code](https://github.com/openshift-pipelines/pipelines-as-code)
+- [Remote Resource Resolution](0060-remote-resource-resolution.md)
+
+Each of these projects are dependent on information from
+repositories - either querying for metadata, fetching content, listening for
+events, recording relatational data about them or publishing notifications to
+them.
+
+Given a shared interest in this data across multiple projects it might
+make sense to define an object that allows an operator to specify common
+repo configuration only once, rather than per-project. Over time other projects
+and tools could also leverage this information.
+
+## Motivation
+
+### Goals
+
+- Decide if there is a common set of data that would be useful across
+  multiple projects.
+- Design an object format to record that common data.
+- Publish developer-facing docs for projects to use.
+- Publish user-facing docs for operators/cluster admins/app teams to use.
+
+### Non-Goals
+
+- TODO: as part of this TEP review let's figure out the scope and what
+  we'd like to keep as non-goals.
+
+### Use Cases
```

Comment:
Would it be possible to separate the use cases from the proposed implementation (a CRD)? I'm guessing you've explored other options as well and it seems like you may have settled on a CRD as the best one. I think it'll be easier to understand why that is if the use cases/requirements don't refer to a CRD-- e.g. it sounds like it would be hard to implement "component could observe Repository objects" without a CRD but "for use as a source of Pipelines and Tasks" could be done via a configmap.

Reply:
Done! I've renamed the TEP to "Common Repository Configuration" and removed mention of CRDs.